src/main/java/com/internet/shop/dao/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,228 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.exceptions.DataBaseProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Role;
+import com.internet.shop.models.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        User user = new User();
+        Set<Role> roles = new HashSet<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM users"
+                    + " JOIN user_roles ON user_roles.user_id = users.user_id"
+                    + " JOIN roles ON  roles.role_id = user_roles.role_id"
+                    + " WHERE user_login = ? AND deleted = false";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, login);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                user = setUserFields(resultSet);
+                roles.add(getRoleWichId(resultSet));
+            }
+
+            while (resultSet.next()) {
+                roles.add(getRoleWichId(resultSet));
+                user.setRoles(roles);
+            }
+            user.setRoles(roles);
+            return Optional.of(user);
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("User by Login " + login
+                    + "is not found", e);
+        }
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (user_name, user_login, user_pass) "
+                + "VALUES (?, ?, ?)";
+        try (PreparedStatement preparedStatement
+                     = ConnectionUtil.getConnection().prepareStatement(query,
+                Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, user.getName());
+            preparedStatement.setString(2, user.getLogin());
+            preparedStatement.setString(3, user.getPassword());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("User is not create", e);
+        }
+        setIdToRoles(user.getRoles());
+        addRolesToUser(user);
+        return user;
+    }
+
+    private void setIdToRoles(Set<Role> roles) {
```

Comment:
Private methods to the end of the class

Reply:
remove private methods in the end of file